synapse/storage/util/id_generators.py

```diff
@@ -352,12 +356,39 @@ def _mark_id_as_finished(self, next_id: int):
 
         with self._lock:
             self._unfinished_ids.discard(next_id)
+            self._finished_ids.add(next_id)
+
+            new_cur = None
+
+            if self._unfinished_ids:
+                # If there are unfinished IDs then the new position will be the
+                # largest finished ID less than the minimum unfinished ID.
+
+                finished = set()
+
+                min_unfinshed = min(self._unfinished_ids)
+                for s in self._finished_ids:
+                    if s < min_unfinshed:
+                        if not new_cur or new_cur < s:
```

Comment:
We do not expect `new_cur` to ever be 0, correct?

Reply:
We do not, though probably clearer to put `is not None`